practice/src/js/templates/productTemplate.js

```diff
@@ -76,4 +88,51 @@ export default class ProductTemplate {
     </div>
   `;
   }
+
+  // Show the delete product modal and set the dataset for confirmation
+  showDeleteModal = (productId) => {
+    const deleteModal = document.getElementById('delete-product-modal');
+    deleteModal.classList.toggle('hidden');
+
+    const confirmBtn = document.getElementById('confirm-btn-delete');
+    confirmBtn.dataset.productId = productId;
+  };
+
+  // Bind events for delete modal confirmation and cancellation
+  bindDeleteModalEvents = (handleConfirmDelete) => {
+    const confirmBtn = document.getElementById('confirm-btn-delete');
+    const cancelBtn = document.getElementById('cancel-btn-delete');
+
+    confirmBtn.addEventListener('click', async () => {
+      const id = confirmBtn.dataset.productId;
+      await handleConfirmDelete(id);
+    });
+
+    cancelBtn.addEventListener('click', () => {
+      this.showDeleteModal();
```

Comment:
The function name is confusing when clicking the cancel button, please update it.